src/services/measure-handler.ts

```diff
@@ -91,335 +102,373 @@ function createExtractor(
   if (extractor.isSW2Format) {
+    if (extractor.descriptionColumns.length < SUPPORTED_LOCALES.length) {
+      throw new FileValidationException(
+        'errors.measure_validation.missing_languages',
+        FileValidationErrorType.MissingLanguages
+      );
+    }
     for (const locale of SUPPORTED_LOCALES) {
       if (extractor.notesColumns) {
-        const notesCol = extractor.notesColumns.find((col) => col.lang === locale.split('-')[0])?.name;
+        const notesCol = extractor.notesColumns.find((col) => col.lang === locale.toLowerCase())?.name;
         if (notesCol) {
-          notesColumnDef = `"${notesCol}" as notes,`;
+          notesColumnDef = `"${notesCol}"`;
         }
       }
       viewComponents.push(
         `SELECT
-                    "${joinColumn}" as reference,
-                    '${locale.toLowerCase()}' AS language,
-                    "${extractor.descriptionColumns.find((col) => col.lang === locale.split('-')[0])?.name}" AS description,
-                    ${notesColumnDef}
-                    ${sortOrderDef}
-                    ${formatColumn} AS format,
-                    ${decimalColumnDef}
-                    ${measureTypeDef}
-                FROM ${lookupTable}\n`
+            "${joinColumn}" AS reference,
+            '${locale.toLowerCase()}' AS language,
+            "${extractor.descriptionColumns.find((col) => col.lang === locale.toLowerCase())?.name}" AS description,
+            ${notesColumnDef} AS notes,
+            ${sortOrderDef} AS sort_order,
+            ${formatColumn} AS format,
+            ${decimalColumnDef} AS decimals,
+            ${measureTypeDef} AS measure_type,
+            ${hierarchyDef} AS hierarchy
+         FROM ${lookupTable}\n`
       );
     }
-    buildMeasureViewQuery = `${viewComponents.join('\nUNION\n')};`;
+    buildMeasureViewQuery = `${viewComponents.join('\nUNION\n')}`;
     logger.debug(`Extracting SW2 measure lookup table to measure table using query ${buildMeasureViewQuery}`);
   } else {
-    if (extractor.notesColumns) {
-      notesColumnDef = `"${extractor.notesColumns[0]}" as notes,`;
+    logger.debug(`Extractor = ${JSON.stringify(extractor, null, 2)}`);
+    if (extractor.notesColumns && extractor.notesColumns.length > 0) {
+      notesColumnDef = `"${extractor.notesColumns[0].name}"`;
+    } else {
+      notesColumnDef = 'NULL';
     }
+
+    const measureMatcher: string[] = [];
+    SUPPORTED_LOCALES.map((locale) => {
+      const lang = locale.split('-')[0].toLowerCase();
+      const tLang = lang;
+      measureMatcher.push(`WHEN LOWER("${extractor.languageColumn}") LIKE '${lang}%' THEN '${locale.toLowerCase()}'`);
+      SUPPORTED_LOCALES.map((locale) => {
+        const lang = locale.split('-')[0].toLowerCase();
+        measureMatcher.push(
+          `WHEN LOWER("${extractor.languageColumn}") LIKE '%${t(`language.${lang}`, { lng: tLang }).toLowerCase()}%' THEN '${locale.toLowerCase()}'`
+        );
+      });
+    });
```

Comment:
Not sure I follow this - it's mapping over supported locales twice?

Reply:
Yes... But it works to produce this case statement:
```
 CASE
WHEN LOWER("lang") LIKE 'en%' THEN 'en-gb' #<-- English
WHEN LOWER("lang") LIKE '%english%' THEN 'en-gb' #<-- Names in English
WHEN LOWER("lang") LIKE '%welsh%' THEN 'cy-gb'
WHEN LOWER("iaith") LIKE 'cy%' THEN 'cy-gb' #<-- Welsh
WHEN LOWER("lang") LIKE '%saesneg%' THEN 'en-gb' #<-- Names in Welsh
WHEN LOWER("lang") LIKE '%cymraeg%' THEN 'cy-gb'
END AS language,
```
The first pass does everything in one language, the subsequent pass does it again for the next language.